streaming/src/main/scala/org/apache/spark/streaming/DStreamGraph.scala

```diff
@@ -167,7 +167,8 @@ final private[streaming] class DStreamGraph extends Serializable with Logging {
    * safe remember duration which can be used to perform cleanup operations.
    */
   def getMaxInputStreamRememberDuration(): Duration = {
-    inputStreams.map { _.rememberDuration }.maxBy { _.milliseconds }
+    // If an InputDStream is not used, its `rememberDuration` will be null and we can ignore them
+    inputStreams.filter(_.rememberDuration != null).map(_.rememberDuration).maxBy(_.milliseconds)
```

Comment:
Simpler to map then filter since you can then just `.filter(_ != null)`? 
Also does this create a corner case where none are non-null? do you need to map null to a default of 0 or something or otherwise handle this?